src/jest-dom/to-have-style.js

```diff
@@ -0,0 +1,79 @@
+import { checkHtmlElement } from '@testing-library/jest-dom/dist/utils';
+
+function getStyleDeclaration(document, css) {
+  const styles = {};
+  const { getComputedStyle } = document.defaultView;
+  // Necessary to normalize colors
+  const copy = document.createElement('div');
+  // Has to be in the document for styles to be applied
+  document.body.append(copy);
+  Object.keys(css).forEach((property) => {
+    copy.style[property] = css[property];
+    styles[property] = getComputedStyle(copy)[property] || css[property];
+  });
+  copy.remove();
+  return styles;
+}
+
+function isSubset(styles, computedStyle) {
+  return (
+    !!Object.keys(styles).length &&
+    Object.entries(styles).every(
+      ([prop, value]) =>
+        computedStyle[prop] === value ||
+        computedStyle.getPropertyValue(prop.toLowerCase()) === value,
+    )
+  );
+}
+
+function printoutStyles(styles) {
+  return Object.keys(styles)
+    .sort()
+    .map((prop) => `${prop}: ${styles[prop]};`)
+    .join('\n');
+}
+
+/** @this {jest.MatcherUtils} */
+export function toHaveStyle(htmlElement, css) {
```

Comment:
I copy-pasted this implementation and made a bunch of changes because the original implementation had dependencies which didn't want to bundle